src/components/Game/Game.js

```diff
@@ -94,34 +88,36 @@ class GameView extends Component {
 
     onSubmit = () => {
         const { lettersInSlots } = this.state;
-        const { dispatch } = this.props;
+        const { dispatch, history } = this.props;
         dispatch(submitAnswer(this.joinLetters(lettersInSlots)));
+        history.replace('/game/result');
     };
 
     onCancel = () => {
         this.setState({
-            submitVisible: false,
+            isSubmitVisible: false,
         });
     };
 
     joinLetters = (arr) => arr.reduce((a, b) => a + b.value, '');
 
     render() {
-        const { answer } = this.props;
-        const { letters, lettersInSlots, submitVisible } = this.state;
-        const lettersOrSubmit = !submitVisible
-            ? (
-                <Letters
-                    letters={letters}
-                    clicked={this.handleLetterClick}
-                />
-            )
-            : (
+        const { letters, lettersInSlots, isSubmitVisible } = this.state;
+
+        let lettersOrSubmit = (
+            <Letters
+                letters={letters}
+                clicked={this.handleLetterClick}
+            />
+        );
+        if (isSubmitVisible) {
+            lettersOrSubmit = (
```

Comment:
I would say it's a bad practice to assign something to variable and then conditionally reassign when it's possible to express it with ternary operator

reassigning makes reasoning about the code a bit harder, you have to read more code to understand what's going on

Reply:
👍